Source/JavaScriptCore/runtime/RegExp.cpp

```diff
@@ -420,55 +420,80 @@ void RegExp::matchCompareWithInterpreter(const String& s, int startOffset, int*
 #endif
 
 #if ENABLE(REGEXP_TRACING)
-    void RegExp::printTraceData()
-    {
-        char formattedPattern[41];
-        char rawPattern[41];
+void RegExp::printTraceHeader()
+{
+    dataLogF("\nRegExp Tracing\n");
+    dataLogF("Regular Expression");
+    for (unsigned i = 0; i < SameLineFormatedRegExpnWidth - 16; ++i)
+        dataLogF(" ");
+    dataLogF("      8 Bit          16 Bit        match()    Matches    Average\n");
+    dataLogF(" <Match only / Match>");
+    for (unsigned i = 0; i < RegExp::SameLineFormatedRegExpnWidth - 21; ++i)
+        dataLogF(" ");
+    dataLogF("      JIT Addr      JIT Address       calls      found   String len\n");
+    for (unsigned i = 0; i < RegExp::SameLineFormatedRegExpnWidth; ++i)
+        dataLogF("-");
+
+    dataLogF("+----------------+----------------+----------+----------+-----------\n");
+}
 
-        strncpy(rawPattern, pattern().utf8().data(), 40);
-        rawPattern[40]= '\0';
+void RegExp::printTraceData()
+{
+    char formattedRegExp[SameLineFormatedRegExpnWidth + 1];
+    char rawPattern[SameLineFormatedRegExpnWidth + 1];
 
-        int pattLen = strlen(rawPattern);
+    strncpy(rawPattern, pattern().utf8().data(), SameLineFormatedRegExpnWidth);
+    rawPattern[SameLineFormatedRegExpnWidth]= '\0';
 
-        snprintf(formattedPattern, 41, (pattLen <= 38) ? "/%.38s/" : "/%.36s...", rawPattern);
+    unsigned pattLen = strlen(rawPattern);
+    unsigned flagsLen = strlen(Yarr::flagsString(flags()).data());
+
+    if (pattLen + flagsLen + 1 < SameLineFormatedRegExpnWidth)
+        snprintf(formattedRegExp, SameLineFormatedRegExpnWidth + 1, "/%s/%s", rawPattern, Yarr::flagsString(flags()).data());
```

Comment:
Let's use makeString and avoid using snprintf.

Reply:
I restructured the code to use makeString() including were we need to convert pointers to hex and right justify.